File: typescript/examples/08-statute-retrieve.ts
/**
 * Example 08 -- Pull a statute section by Bluebook citation
 * Language: TypeScript (Node.js)
 *
 * Requires a proper section number (e.g. 42 U.S.C. § 1983). Cite-check confirms
 * existence; this call delivers statute.body when LawDiver holds or can fetch it.
 */
import { LawDiverClient } from "../src/client.js";

const client = new LawDiverClient();
const query = process.argv[2] ?? "42 U.S.C. § 1983";

const hit = await client.retrieveStatute({ query });

console.log(`status: ${hit.status}`);
console.log(`message: ${hit.message ?? ""}`);
console.log(`requestId: ${hit.requestId}`);

if (hit.status === "ok" && hit.statute) {
  console.log(`bluebook: ${hit.statute.bluebook}`);
  console.log(`heading:  ${hit.statute.heading ?? "(none)"}`);
  console.log(`chars:    ${hit.statute.bodyChars ?? 0}`);
  console.log(`key:      ${hit.statute.authorityKey}`);
  const preview = (hit.statute.body ?? "").slice(0, 280).replace(/\s+/g, " ").trim();
  console.log(`preview:  ${preview}${(hit.statute.body?.length ?? 0) > 280 ? "…" : ""}`);
} else if (hit.status === "not_found") {
  console.log("No such section.", hit.note ?? "");
} else if (hit.status === "not_a_statute") {
  console.log("Not a statute citation. Include jurisdiction + section number.");
} else {
  console.log("Unavailable.", hit.note ?? hit.message ?? "");
  if (hit.statute?.officialUrl) console.log(`officialUrl: ${hit.statute.officialUrl}`);
}

File: python/examples/08_statute_retrieve.py
"""Example 08 -- Pull a statute section by Bluebook citation.

Requires a proper section number (e.g. 42 U.S.C. § 1983).
"""

from __future__ import annotations

import sys

from lawdiver import LawDiverClient


def main() -> None:
    query = sys.argv[1] if len(sys.argv) > 1 else "42 U.S.C. § 1983"
    client = LawDiverClient()
    hit = client.retrieve_statute(query=query)

    print(f"status: {hit.get('status')}")
    print(f"message: {hit.get('message') or ''}")
    print(f"requestId: {hit.get('requestId')}")

    statute = hit.get("statute") or {}
    if hit.get("status") == "ok" and statute:
        print(f"bluebook: {statute.get('bluebook')}")
        print(f"heading:  {statute.get('heading') or '(none)'}")
        print(f"chars:    {statute.get('bodyChars') or 0}")
        print(f"key:      {statute.get('authorityKey')}")
        body = statute.get("body") or ""
        preview = " ".join(body[:280].split())
        suffix = "…" if len(body) > 280 else ""
        print(f"preview:  {preview}{suffix}")
    elif hit.get("status") == "not_found":
        print("No such section.", hit.get("note") or "")
    elif hit.get("status") == "not_a_statute":
        print("Not a statute citation. Include jurisdiction + section number.")
    else:
        print("Unavailable.", hit.get("note") or hit.get("message") or "")
        if statute.get("officialUrl"):
            print(f"officialUrl: {statute.get('officialUrl')}")


if __name__ == "__main__":
    main()
